XamarinCommunityToolkit/Views/TabView/MaterialTabViewItemTemplate.shared.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	[Preserve(AllMembers = true)]
+	public class MaterialTabViewItemTemplate : Grid
+	{
+		readonly VisualFeedbackEffect visualFeedback;
+
+		readonly Image icon;
+		readonly Label text;
+		readonly TabBadgeView badge;
+
+		public MaterialTabViewItemTemplate()
+		{
+			visualFeedback = new VisualFeedbackEffect();
+			Effects.Add(visualFeedback);
+
+			RowSpacing = 0;
+
+			HorizontalOptions = LayoutOptions.FillAndExpand;
+			VerticalOptions = LayoutOptions.FillAndExpand;
+
+			RowDefinitions.Add(new RowDefinition { Height = GridLength.Star });
+			RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
+
+			icon = new Image
+			{
+				Aspect = Aspect.AspectFit,
+				HorizontalOptions = LayoutOptions.Center,
+				VerticalOptions = LayoutOptions.Center,
+				Margin = new Thickness(0, 6)
+			};
+
+			text = new Label
+			{
+				FontSize = Device.GetNamedSize(NamedSize.Medium, typeof(Label)),
+				FontAttributes = FontAttributes.Bold,
+				HorizontalOptions = LayoutOptions.Center,
+				VerticalOptions = LayoutOptions.Center,
+				Margin = new Thickness(0, 6)
+			};
+
+			badge = new TabBadgeView
+			{
+				PlacementTarget = icon,
+				Margin = new Thickness(0)
+			};
+
+			Children.Add(icon);
+			Children.Add(text);
+			Children.Add(badge);
+
+			SetRow(icon, 0);
+			SetRow(text, 1);
+			SetRow(badge, 0);
+			SetRowSpan(badge, 2);
+		}
+
+		protected override void OnParentSet()
+		{
+			base.OnParentSet();
+
+			BindingContext = Parent;
+
+			icon.SetBinding(Image.SourceProperty, "CurrentIcon");
+
+			text.SetBinding(Label.TextProperty, "Text", BindingMode.OneWay, new ToUpperConverter());
```

Comment:
Can we replace this with the one [we have](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/XamarinCommunityToolkit/Converters/TextCaseConverter.shared.cs)? 

That saves us one duplicate and the tests are then fixed!

Reply:
Sure, changed!